src/Viz.js

```diff
@@ -423,10 +424,24 @@ export default class Viz extends BaseClass {
       if (!this._height) this.height(h);
     }
 
-    this._select.transition(this._transition)
+    this._select
+      .transition(this._transition)
+      .attr("aria-labelledby", `${this._uuid}-title ${this._uuid}-desc`)
+      .attr("role", "img")
       .style("width", `${this._width}px`)
       .style("height", `${this._height}px`);
 
+    // Updates the <title> tag if already exists else creates a new <title> tag on this.select.
+    const svgTitleText = this._title || "D3plus Visualization";
+    const svgTitle = this._select.selectAll("title").data([0]);
+    const svgTitleEnter = svgTitle.enter().append("title");
+    svgTitle.merge(svgTitleEnter).text(svgTitleText).attr("id", `${this._uuid}-title`);
+
+    // Updates the <desc> tag if already exists else creates a new <desc> tag on this.select.
+    const svgDesc = this._select.selectAll("desc").data([0]);
+    const svgDescEnter = svgDesc.enter().append("desc");
+    svgDesc.merge(svgDescEnter).text(this._svgDesc).attr("id", `${this._uuid}-desc`);
```

Comment:
Great job with these enter/update functions. One small tweak that would make this more precise: setting the `"id"` attribute only needs to occur on enter, as the uuid does not change on subsequent renders. You can move the `.attr("id", `${this._uuid}-desc`)` code for both the title and desc up 1 line to where you append the item.